### PR TITLE
JValue.asInt/getInt methods #34

### PR DESCRIPTION
**Pull request by tapmania**

### CODE REVIEW

**Comment by non** [comment]
Thanks! I have one question but otherwise this looks good.

File: ast/src/main/scala/jawn/ast/JValue.scala [hunk 1]
```diff
@@ -143,7 +145,7 @@ object JNum { self =>
 
   final def hybridEq(x: Long, y: Double): Boolean = {
     val z = x.toDouble
-    y == z && z.toLong == x
+    y == z && y.toLong == x
```
Comment (tapmania) [inline ast/src/main/scala/jawn/ast/JValue.scala]: Intuitively it seems like z.ToLong should always be equal to x as z was created from x. I thought a better comparison would be to compare it to y.toLong. Please correct me if I am wrong.

Comment (non) [inline ast/src/main/scala/jawn/ast/JValue.scala]: So, the idea here is that there are large `Long` values which can't be exactly represented as a `Double`. Here's an example:

```
scala> val x: Long = (1L << 55) + 1L
x: Long = 36028797018963969

scala> val z = x.toDouble
z: Double = 3.6028797018963968E16

scala> z.toLong
res0: Long = 36028797018963968

scala> z.toLong == x
res1: Boolean = false
```

Normally when you compare a `Long` and a `Double` in Scala, Scala will convert the `Long` into a `Double` first. This means that a `Long` will be equal to the "closest" `Double`, even if they aren't exactly the same. By doing `z.toLong == x` we are ensuring that the `Double` does convert back to the original `Long`. If this fails it means the `Long` was too bad (as in the REPL example) and so the comparison should fail.

Does that make sense? It seems like I should have added a comment explaining what is going on there.

Comment (tapmania) [inline ast/src/main/scala/jawn/ast/JValue.scala]: @non thx for the clear explanation! I understand now. do you want me to revert my change then?

Comment (non) [inline ast/src/main/scala/jawn/ast/JValue.scala]: Yes, if you don't mind. Otherwise it the PR is great.

Comment (tapmania) [inline ast/src/main/scala/jawn/ast/JValue.scala]: @non done!

**Comment by non** [comment]
Thanks!

**Pull request merged**